### ISSUE TITLE
Support for ECDH-ES-* on Linux is possible

### ISSUE THREAD

**Issue by polewskm** [dvsekhvalnov/jose-jwt#219]
See the following:
https://github.com/polewskm/jose-jwt-ecdh

**Comment by dvsekhvalnov** [comment]
Hi @polewskm , 
thanks for snippet. Is it actually working under linux? I don't see calls to `Encode`\`Decode` ?

Assuming the code is for derivation part here: https://github.com/dvsekhvalnov/jose-jwt/blob/master/jose-jwt/jwa/EcdhKeyManagement.cs#L92 ?

**Comment by polewskm** [comment]
Yes, my implementation is working on Linux. To be clear, you would have to:
- Refactor your `ConcatKDF.DeriveKey` to NOT use `ECDiffieHellmanCng` (notice the `...Cng` suffix here)
- There is no equivalent implementation of `SP800_56A_CONCAT` in Linux
- You have to implement the functionality of `SP800_56A_CONCAT` yourself
- That is what my snippet is showing, how to manually implement `SP800_56A_CONCAT` on Linux using `ECDiffieHellman` (notice that lack of `...Cng` in the suffix here)

**Comment by dvsekhvalnov** [comment]
Sp800_56a is easy to bring, have it somewhere in Go version of lib.

So if you have working version, mind PR then? :)

**Comment by polewskm** [comment]
Sure can do. It might be a week or two before I submit a PR though.

**Comment by dvsekhvalnov** [comment]
Will greatly appreciate, thank you.

**Comment by deepak-sidhyati** [comment]
@dvsekhvalnov @polewskm is there a PR for this?

**Comment by dvsekhvalnov** [comment]
@deepak-sidhyati , haven't seen PR so far. 

Can may be find some time to port it over myself, but can't commit on timeline.

**Comment by polewskm** [comment]
Sorry, I have been terrible busy with other projects and haven't found the time to work on this yet.

**Comment by digaomatias** [comment]
Looking forward to this one getting ready to be used as well.

**Comment by digaomatias** [comment]
> Looking forward to this one getting ready to be used as well.

Just want to let you know. I got the code snippet posted by @polewskm and followed his instructions on what should be done. I was able to run the existing unit test for the ConcatKDF class but invoking my method (@polewskm's code) and it passed! Which is great. I'll change the code that uses the ConcatKDF to invoke this new method and see if the unit tests pass overall. I'll come up with a PR in the next few days and you guys can review if it makes sense or not.
